src/cloudai/schema/test_template/jax_toolbox/kubernetes_job_spec_gen_strategy.py

```diff
@@ -0,0 +1,224 @@
+                                            ),
+                                        ],
+                                        "resources": self._prepare_launcher_resources(),
+                                    }
+                                ],
+                                "restartPolicy": "Never",
+                            },
+                        },
+                    },
+                    "Worker": {
+                        "replicas": final_num_nodes,
+                        "template": {
+                            "spec": {
+                                "containers": [
+                                    {
+                                        "image": cmd_args["docker_image_url"],
+                                        "name": "jax-toolbox-worker",
+                                        "env": self._generate_env_list(env_vars),
+                                        "command": ["/bin/bash"],
+                                        "args": ["-c", "/usr/sbin/sshd -p 2222; sleep infinity"],
+                                        "resources": self._prepare_worker_resources(),
+                                        "volumeMounts": [
+                                            {"mountPath": "/dev/shm", "name": "dshm"},
+                                        ],
+                                    }
+                                ],
+                                "volumes": [{"name": "dshm", "emptyDir": {"medium": "Memory"}}],
+                            },
+                        },
+                    },
+                },
+            },
+        }
+
+        return job_spec
+
+    def _generate_launcher_command(
+        self,
+        final_num_nodes: int,
+        nodes: List[str],
+        env_vars: Dict[str, str],
+        cmd_args: Dict[str, str],
+        extra_cmd_args: str,
+    ) -> str:
+        """Generate the launcher command for Kubernetes."""
+        pretest_command = self._generate_pretest_command(env_vars, cmd_args)
+        main_command = self._generate_main_command(env_vars, cmd_args, extra_cmd_args)
+        output_file = f"{cmd_args['output_path']}/output_pretest-*.txt"
+        keyword = cmd_args.get("keyword", "Avg bus bandwidth")
+
+        return f"""
+            # Run pretest command
+            {pretest_command}
+
+            # Check the output of pretest for keyword
+            file_pattern="{output_file}"
+            keyword="{keyword}"
+            keyword_found=false
+
+            if grep -q "$keyword" $file_pattern; then
+                keyword_found=true
+            fi
+
+            # Conditionally run main command
+            if [ "$keyword_found" = true ]; then
+                {main_command}
+            else
+                echo "Keyword not found. Skipping main command."
+            fi
+        """
+
+    def _generate_pretest_command(self, env_vars: Dict[str, str], cmd_args: Dict[str, str]) -> str:
```

Comment:
Do we need the pre-test for initial k8 port? Assuming this will be officially part of the prologue and epilogue support? 

Also if this is going here, this needs to follow all the tricks and optimization you figured out during your NCCL tests on k8.